ui/src/components/Profile.vue

```diff
@@ -0,0 +1,197 @@
+                    ? profileKeys[item].classes
+                    : 'col-lg-4 col-md-6 col-12'
+                "
+                v-if="item !== 'bio' && item !== 'username'"
+              >
+                <q-input
+                  :label="profileKeys[item].label"
+                  :value="value[profileKeys[item].name]"
+                  :disable="profileKeys[item].disable"
+                  :autogrow="profileKeys[item].autogrow"
+                  :mask="profileKeys[item].mask"
+                  @input="(val) => inputHandler(val, item)"
+                />
+              </div>
+            </template>
+          </div>
+        </q-card-section>
+      </q-card-section>
+      <q-card-actions class="q-mb-md">
+        <q-space />
+        <q-btn
+          color="primary"
+          rounded
+          unelevated
+          class="q-px-md __btn-submit"
+          label="Save"
+          :loading="loading && loading.save"
+          :disable="loading && loading.save"
+          @click="handleSaveInfo"
+        >
+          <template #loading>
+            <q-spinner-dots color="white" />
+          </template>
+        </q-btn>
+        <q-btn
+          unelevated
+          rounded
+          :icon="!expanded ? 'expand_more' : 'expand_less'"
+          color="red-5"
+          class="q-px-md"
+          @click="expanded = !expanded"
+        >
+          <q-tooltip>
+            {{ expanded ? "Hide change password" : "Change Password" }}
+          </q-tooltip>
+        </q-btn>
+      </q-card-actions>
+
+      <q-slide-transition>
+        <div v-show="expanded">
+          <q-separator />
+          <password-change
+            @changePassword="(val) => $emit('changePassword', val)"
+            :loading="loading && loading.changePassword"
+          />
+        </div>
+      </q-slide-transition>
+    </q-card>
+  </div>
+</template>
+
+<script>
+import PasswordChange from "./PasswordChange.vue";
+import {
+  QAvatar,
+  QCard,
+  QCardSection,
+  QCardActions,
+  QBtn,
+  QSpace,
+  QSlideTransition,
+  QSeparator,
+  QFile,
+} from "quasar";
+export default {
+  name: "QProfile",
+  props: {
+    value: {
+      type: Object,
+      required: true,
+    },
+    schema: {
+      type: Array,
+      required: true,
+    },
+    loading: {
+      type: Object,
+      required: false,
+    },
+  },
+  components: {
+    QAvatar,
+    QCard,
+    QCardSection,
+    QCardActions,
+    QBtn,
+    QFile,
+    QSpace,
+    QSlideTransition,
+    QSeparator,
+    PasswordChange,
+  },
+  data: () => ({
+    expanded: false,
+    profileImage: undefined,
+    profileImageBlob: undefined,
+  }),
+  computed: {
+    /** @returns { array } */
+    profileKeys() {
+      const items = {};
+      this.schema.forEach((item) => {
+        items[item.name] = item;
+      });
+
+      return items;
+    },
+  },
+  watch: {
+    profileImage(val) {
+      this.profileImageBlob = URL.createObjectURL(val);
+    },
```

Comment:
It's recommended to revokeObjectURL when we finish using the URL so let's do that..

```suggestion
    profileImage(val, oldVal) {
      if(oldVal) {
        URL.revokeObjectURL(oldVal);
        this.profileImageBlob = null;
      }
      if(val) {
        this.profileImageBlob = URL.createObjectURL(val);
      }
    },
```